src/app/script/app_clipboard_object.cpp

```diff
@@ -0,0 +1,138 @@
+// Aseprite
+// Copyright (c) 2022-2024  Igara Studio S.A.
+//
+// This program is distributed under the terms of
+// the End-User License Agreement for Aseprite.
+
+#ifdef HAVE_CONFIG_H
+  #include "config.h"
+#endif
+
+#include "app/modules/palettes.h"
+#include "app/script/luacpp.h"
+#include "app/util/clipboard.h"
+#include "engine.h"
+#include "clip/clip.h"
+
+namespace app {
+namespace script {
+
+namespace {
+
+struct Clipboard {};
+
+int Clipboard_clear(lua_State* L)
+{
+  if (!clip::clear())
+    return luaL_error(L, "failed to clear the clipboard");
+
+  return 1;
+}
+
+int Clipboard_is_text(lua_State* L)
+{
+  lua_pushboolean(L, clip::has(clip::text_format()));
+  return 1;
+}
+
+int Clipboard_is_image(lua_State* L)
+{
+  lua_pushboolean(L, clip::has(clip::image_format()));
+  return 1;
+}
+
+int Clipboard_is_empty(lua_State* L)
+{
+  // Using clip::has(clip::empty_format()) had inconsistent results, might as well avoid false positives
+  // by just checking the two formats we support
```

Comment:
Reading your comment I'm thinking that probably we should remove this `isEmpty` function/property. This was a point of discussion in a issue from the clip project in the past: https://github.com/dacap/clip/issues/28